arrow-libs/fx/arrow-fx-stm/src/main/kotlin/arrow/fx/stm/STM.kt

```diff
@@ -119,26 +118,24 @@ import arrow.fx.stm.internal.lookupHamtWithHash
  * }
  * //sampleEnd
  *
- * fun main() {
- *   Environment().unsafeRunSync {
- *     val acc1 = TVar.new(0)
- *     val acc2 = TVar.new(300)
- *     println("Balance account 1: ${acc1.unsafeRead()}")
- *     println("Balance account 2: ${acc2.unsafeRead()}")
- *     ForkConnected {
- *       println("Sending money - Searching")
- *       sleep(2.seconds)
- *       println("Sending money - Found some")
- *       atomically { acc1.write(100_000_000) }
- *     }
- *     println("Performing transaction")
- *     atomically {
- *       println("Trying to transfer")
- *       transfer(acc1, acc2, 50)
- *     }
- *     println("Balance account 1: ${acc1.unsafeRead()}")
- *     println("Balance account 2: ${acc2.unsafeRead()}")
+ * fun main(): Unit = runBlocking {
```

Comment:
Does it make a difference here to use the suspended main function?

Reply:
Yes, `runBlocking` spins up an `EventLoop` `CoroutineContext` which allows you to go back to the original `CoroutineContext`.

For example:

```
suspend fun printThread() : Unit = println(Thread.currentThread().name)

suspend fun main() {
    printThread() //main
    withContext(Dispatchers.IO) {
        printThread() // DefaultDispatcher-worker-1
    }
    printThread() // DefaultDispatcher-worker-1
}
```

VS

```
suspend fun main() = runBlocking<Unit> {
    printThread() //main
    withContext(Dispatchers.IO) {
        printThread() // DefaultDispatcher-worker-1
    }
    printThread() // main
}
```

Reply:
Thank you